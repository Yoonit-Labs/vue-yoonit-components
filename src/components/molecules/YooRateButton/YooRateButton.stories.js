import YooRateButton from './YooRateButton.vue'
import PropsConfig from './YooRateButton.config'

export default {
  title: 'Yoonit/YooRateButton',
  component: YooRateButton,
  argTypes: {
    size: {
      control: {
        type: 'select',
        options: PropsConfig.size.options
      }
    }
  }
}

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { YooRateButton },
  template:
    '<yoo-rate v-bind="$props"></yoo-rate>'
})

export const Custom = Template.bind({})
Custom.args = { text: 'Yoo Switch' }