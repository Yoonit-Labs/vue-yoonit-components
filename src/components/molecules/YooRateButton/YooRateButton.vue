<template lang="pug">
  YooFlexLayout(
    class="yoo-rate"
    justifyContent="space-around"
    alignItems="center"
  )
    YooButton(
     v-for="(item, index) in rate"
     :key="index"
     :text="item.text"
     :fill="selectedIn === index ? 'primary' : 'light'"
     buttonSize="xs"
     variation="base"
     @onClick="doActive(index, item.value)"
    )

</template>

<script>

import YooFlexLayout from '@/components/quarks/YooFlexLayout/YooFlexLayout.vue'

import YooButton from '@/components/atoms/YooButton/YooButton.vue'

export default {
  name: 'YooRateButton',
  components: {
    YooFlexLayout,
    YooButton
  },
  props: {
    rate: {
      type: Array,
      required: true
    }
  },
  data: () => ({
    selectedIn: ''
  }),
  beforeCreate () {},
  created () {},
  beforeMount () {},
  mounted () {},
  beforeUpdate () {},
  updated () {},
  beforeDestroy () {},
  destroyed () {},
  computed: {},
  methods: {
    doActive (index, valor) {
      this.$emit('doClick', valor)
      this.selectedIn = index
    }
  },
  filters: {},
  watch: {}
}
</script>

<style src="./YooCard.sass" lang="sass" scoped></style>