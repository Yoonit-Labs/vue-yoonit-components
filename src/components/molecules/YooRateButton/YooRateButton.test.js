import { shallowMount } from '@vue/test-utils'
import YooRateButton from '@/components/molecules/YooRateButton/YooRateButton.vue'

const classBlock = 'yoo-rate'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooRateButton, {
    slots: { default: SlotText },
    propsData: {
      rate: [
        {
          text: '1',
          value: 1
        },
        {
          text: '2',
          value: 2
        },
        {
          text: '3',
          value: 3
        }
      ]
    }
  })
}

describe('YooRateButton Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', async () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.classes('yoo-rate')).toBe(true)
    expect(wrapper.find('button.yoo-btn').exists()).toBe(true)
  })

  describe('Events', () => {
    describe('doCheck', () => {
      it('Emits doCheck Event', async () => {
        wrapper.find('.yoo-rate').trigger('click')
        expect(wrapper.emitted()).toHaveProperty('response')
      })
    })
  }) // describe Events
})
