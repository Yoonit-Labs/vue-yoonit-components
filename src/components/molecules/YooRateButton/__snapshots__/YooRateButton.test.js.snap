// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooRateButton Component Matches Snapshot 1`] = `
<yooflexlayout-stub flexdirection="row" flexwrap="nowrap" justifycontent="space-around" alignitems="center" aligncontent="stretch" gap="0" class="yoo-rate">
  <yoobutton-stub variation="base" buttonsize="xs" fill="light" icon="" iconposition="left" iconsize="normal" iconstyle="solid" animation="none" text="1" iconcustomcolor=""></yoobutton-stub>
  <yoobutton-stub variation="base" buttonsize="xs" fill="light" icon="" iconposition="left" iconsize="normal" iconstyle="solid" animation="none" text="2" iconcustomcolor=""></yoobutton-stub>
  <yoobutton-stub variation="base" buttonsize="xs" fill="light" icon="" iconposition="left" iconsize="normal" iconstyle="solid" animation="none" text="3" iconcustomcolor=""></yoobutton-stub>
</yooflexlayout-stub>
`;
